Improve fdas (#236)
telefonicaid/fiware-data-access: doc/02_architecture.md

-   **refreshPolicy**: object defining automatic refresh behaviour (`none`, `interval`, or `window`)
-   **status**: current execution status (`fetching`, `transforming`, `uploading`, `completed`, `failed`)
-   **progress**: execution progress percentage (0–100)
-   **initFetch**: timestamp of the current/last fetch start (ISO date)
-   **lastFetch**: timestamp of the last successful fetch completion (ISO date)
-   **datasourceId**: datasource identifier used to resolve source credentials (default `default` when omitted)
-   **validationMode**: validation mode (`strict` or `unchecked`, default `strict`)

> -   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only)

Reviewer: Maybe we could add:

```suggestion
-   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only). By the moment, only for FDA based in PG datasources (for other datasources se have to consider how to it applies)
```

Maybe even including a reference to the issue number in which we are discussing this.

Author: I am not sure if we should add the Issue reference. It is a bit complicated because it is present in multiple issues:
- Metadata in Mongo FDAs collection: https://github.com/telefonicaid/fiware-data-access/issues/233
- Metadata for FDAs with datasourse MongoDB: https://github.com/telefonicaid/fiware-data-access/issues/235
- Schema metadata Issue: https://github.com/telefonicaid/fiware-data-access/issues/237

Reviewer: I think the key one (with regards to evolution of the `schema` field in the case of schemaless datasources) is https://github.com/telefonicaid/fiware-data-access/issues/235

> -   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only). By the moment, only for FDA based in PG datasources (for other datasources se have to consider how to it applies)

Reviewer: ```suggestion
-   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only). It is an optional field, by the moment used only in FDAs based in PG datasources (for other datasources se have to consider how to it applies, specially for schema-less ones, see issue [#235](https://github.com/telefonicaid/fiware-data-access/issues/235))
```

> -   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only). It is an optional field, by the moment used only in FDAs based in PG datasources (for other datasources se have to consider how to it applies, specially for schema-less ones, see issue [#235](https://github.com/telefonicaid/fiware-data-access/issues/235))

Reviewer: Typo

```suggestion
-   **schema**: array of column definitions (`name` and `type`) persisted from the source schema (`strict` mode only). It is an optional field, by the moment used only in FDAs based in PG datasources (for other datasources we have to consider how to it applies, specially for schema-less ones, see issue [#235](https://github.com/telefonicaid/fiware-data-access/issues/235))
```

-   **refreshPolicy**: object defining automatic refresh behaviour (`none`, `interval`, or `window`)
-   **status**: current execution status (`fetching`, `transforming`, `uploading`, `completed`, `failed`)
-   **progress**: execution progress percentage (0–100)

> -   **initFetch**: timestamp of the current/last fetch start (ISO date)

Reviewer: What about the schema? I understand it is also persisted in MongoDB

Author: Yes, it should be added

Author: Fixed in https://github.com/telefonicaid/fiware-data-access/pull/236/commits/9b3f3444ff4b38c026e8e726b6a6360bbf7ef737
